adam-cli/src/main/scala/org/bdgenomics/adam/cli/Vcf2Adam.scala

```diff
@@ -0,0 +1,126 @@
+  var outputPath: String = null
+
+  @Argument(required = true, metaVar = "VARIANTS", usage = "Location to write ADAM variants data in Parquet format.", index = 1)
+  var variantsOutputPath: String = null
+
+  @Argument(required = true, metaVar = "GENOTYPES", usage = "Location to write ADAM genotypes data in Parquet format.", index = 2)
+  var genotypesOutputPath: String = null
+
+  @Args4jOption(required = false, name = "-coalesce", usage = "Number of partitions written to the ADAM output directories.")
+  var coalesce: Int = -1
+
+  @Args4jOption(required = false, name = "-force_shuffle_coalesce", usage = "Even if the repartitioned RDDs have fewer partitions, force a shuffle.")
+  var forceShuffle: Boolean = false
+
+  @Args4jOption(required = false, name = "-sort_on_save", usage = "Sort by contig index.")
+  var sort: Boolean = false
+
+  @Args4jOption(required = false, name = "-sort_lexicographically_on_save", usage = "Sort by lexicographic order. Conflicts with -sort_on_save.")
+  var sortLexicographically: Boolean = false
+
+  @Args4jOption(required = false, name = "-stringency", usage = "Stringency level for various checks; can be SILENT, LENIENT, or STRICT. Defaults to STRICT.")
+  var stringency: String = "STRICT"
+}
+
+/**
+ * Transform VCF into genotypes and variants in ADAM format.
+ */
+class Vcf2Adam(val args: Vcf2AdamArgs)
+    extends BDGSparkCommand[Vcf2AdamArgs] {
+  val companion = Vcf2Adam
+  val stringency = ValidationStringency.valueOf(args.stringency)
+
+  /**
+   * Coalesce the specified GenomicRDD if requested.
+   *
+   * @param rdd GenomicRDD to coalesce.
+   * @return The specified GenomicRDD coalesced if requested.
+   */
+  private def maybeCoalesce[U <: GenomicRDD[_, U]](rdd: U): U = {
+    if (args.coalesce != -1) {
+      log.info("Coalescing the number of partitions to '%d'".format(args.coalesce))
+      if (args.coalesce > rdd.rdd.partitions.length || args.forceShuffle) {
+        rdd.transform(_.coalesce(args.coalesce, shuffle = true))
+      } else {
+        rdd.transform(_.coalesce(args.coalesce, shuffle = false))
+      }
+    } else {
+      rdd
+    }
+  }
+
+  /**
+   * Sort the specified GenomicRDD if requested.
+   *
+   * @param rdd GenomicRDD to sort.
+   * @return The specified GenomicRDD sorted if requested.
+   */
+  private def maybeSort[U <: GenomicRDD[_, U]](rdd: U): U = {
+    if (args.sort) {
+      log.info("Sorting before saving")
+      rdd.sort()
+    } else if (args.sortLexicographically) {
+      log.info("Sorting lexicographically before saving")
+      rdd.sortLexicographically()
+    } else {
+      rdd
+    }
+  }
+
+  def run(sc: SparkContext) {
+    require(!(args.sort && args.sortLexicographically),
+      "Cannot set both -sort_on_save and -sort_lexicographically_on_save.")
+
+    val variantContexts = sc.loadVcf(
+      args.inputPath,
+      stringency = stringency)
+
+    // todo: cache variantContexts?  sort variantContexts first?
```

Comment:
I would present this as an option. It's fine to convert twice if you're working with a small dataset, but you'll want to persist if you're working with a large data.